src/components/MatrixBuilder/index.js

```diff
@@ -0,0 +1,27 @@
+import { InputBlock } from "../InputBlock";
+import { CustomButton } from "../shared/CustomButton";
+import styles from "./MatrixBuilder.module.scss"
+
+export const MatrixBuilder = ({ data, setData, matrixData, setMatrixData, }) => {
+	const createMatrix = () => {
+		let arr = Array(matrixData.rows).fill().map(() =>
+			Array(matrixData.columns).fill().map(() => ({ id: Math.floor(Math.random() * (999 - 100) + 100), num: Math.floor(Math.random() * (999 - 100) + 100) })));
+		setData({ ...data, arr: arr, numberRows: 1 })
+	}
+
+	const enterData = (event) => {
```

Comment:
change it to handleSetSettings = (event) => {
const {value, name} = event.target
onSetSettings({...settings, name: value})
}
no need to create a matrix on press enter. only with button